NetworkUtilsTests.testNonExistingInterface failures on Ubuntu 20.04 runners

**Build scan**: https://gradle-enterprise.elastic.co/s/ytxjavyusk7gw

**Repro line**: `./gradlew ':server:test' --tests "org.elasticsearch.common.network.NetworkUtilsTests.testNonExistingInterface" -Dtests.seed=3880BFAC1B01C494 -Dtests.security.manager=true -Dtests.locale=sl -Dtests.timezone=SystemV/EST5EDT -Druntime.java=8`

**Reproduces locally?**: No (on MacOS, all failures on CI occurred on `ubuntu-2004-lts` runners)

**Applicable branches**: `master`, `7.x`

**Failure history**: [build stats](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(branch,message,class,test),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'test:testNonExistingInterface'),sort:!(time,desc)))
Not sure why, but Gradle Enterprise shows a couple other failures that don't show up in build stats. First failure was on Dec. 14.

**Failure excerpt**:

```
java.lang.AssertionError: 	
Expected: a string containing "veth219b311:suffix"	
     but: was "setting [settingValue] matched no network interfaces; valid values include [_lo:suffix_, _ens4:suffix_, _docker0:suffix_]"	
	at __randomizedtesting.SeedInfo.seed([3880BFAC1B01C494:EC5BA0A0EA3F45F0]:0)	
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)	
	at org.junit.Assert.assertThat(Assert.java:956)	
	at org.junit.Assert.assertThat(Assert.java:923)	
	at org.elasticsearch.common.network.NetworkUtilsTests.testNonExistingInterface(NetworkUtilsTests.java:102)	
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)	
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	
	at java.lang.reflect.Method.invoke(Method.java:498)	
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)	
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)	
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)	
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)	
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)	
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)	
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)	
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)	
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)	
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)	
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)	
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)	
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)	
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)	
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)	
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)	
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)	
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)	
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)	
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)	
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)	
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)	
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)	
	at java.lang.Thread.run(Thread.java:748)
```

Another one, this time with 7.11 on debian-9: https://gradle-enterprise.elastic.co/s/pd2r7ua7syngy

Error message is:
```
org.elasticsearch.common.network.NetworkUtilsTests > testNonExistingInterface FAILED
    java.lang.AssertionError: 
    Expected: a string containing "vethc93b867:suffix"
         but: was "setting [settingValue] matched no network interfaces; valid values include [_lo:suffix_, _eth0:suffix_, _docker0:suffix_]"
        at __randomizedtesting.SeedInfo.seed([54A88AA05E172FF2:807395ACAF29AE96]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.common.network.NetworkUtilsTests.testNonExistingInterface(NetworkUtilsTests.java:102)
```

Reproduction line: 
> ./gradlew ':server:test' --tests "org.elasticsearch.common.network.NetworkUtilsTests.testNonExistingInterface" -Dtests.seed=54A88AA05E172FF2 -Dtests.security.manager=true -Dtests.locale=hi -Dtests.timezone=Africa/Lusaka -Druntime.java=8

Not reproducible locally (MacOS)

--- a/server/src/test/java/org/elasticsearch/common/network/NetworkUtilsTests.java
+++ b/server/src/test/java/org/elasticsearch/common/network/NetworkUtilsTests.java
@@ -35,7 +35,7 @@ import static org.hamcrest.Matchers.equalTo;
  * Tests for network utils. Please avoid using any methods that cause DNS lookups!
  */
 public class NetworkUtilsTests extends ESTestCase {
-    
+
     /**
      * test sort key order respects PREFER_IPV4
      */
@@ -45,7 +45,7 @@ public class NetworkUtilsTests extends ESTestCase {
         assertTrue(NetworkUtils.sortKey(localhostv4, false) < NetworkUtils.sortKey(localhostv6, false));
         assertTrue(NetworkUtils.sortKey(localhostv6, true) < NetworkUtils.sortKey(localhostv4, true));
     }
-    
+
     /**
      * test ordinary addresses sort before private addresses
      */
@@ -55,14 +55,14 @@ public class NetworkUtilsTests extends ESTestCase {
         InetAddress ordinary = InetAddress.getByName("192.192.192.192");
         assertTrue(NetworkUtils.sortKey(ordinary, true) < NetworkUtils.sortKey(siteLocal, true));
         assertTrue(NetworkUtils.sortKey(ordinary, false) < NetworkUtils.sortKey(siteLocal, false));
-        
+
         InetAddress siteLocal6 = InetAddress.getByName("fec0::1");
         assert siteLocal6.isSiteLocalAddress();
         InetAddress ordinary6 = InetAddress.getByName("fddd::1");
         assertTrue(NetworkUtils.sortKey(ordinary6, true) < NetworkUtils.sortKey(siteLocal6, true));
         assertTrue(NetworkUtils.sortKey(ordinary6, false) < NetworkUtils.sortKey(siteLocal6, false));
     }
-    
+
     /**
      * test private addresses sort before link local addresses
      */
@@ -73,7 +73,7 @@ public class NetworkUtilsTests extends ESTestCase {
         assertTrue(NetworkUtils.sortKey(ordinary, true) < NetworkUtils.sortKey(linkLocal, true));
         assertTrue(NetworkUtils.sortKey(ordinary, false) < NetworkUtils.sortKey(linkLocal, false));
     }
-    
+
     /**
      * Test filtering out ipv4/ipv6 addresses
      */
@@ -99,7 +99,10 @@ public class NetworkUtilsTests extends ESTestCase {
                 () -> NetworkUtils.getAddressesForInterface("settingValue", ":suffix" , "non-existing"));
         assertThat(exception.getMessage(), containsString("setting [settingValue] matched no network interfaces; valid values include"));
         for (NetworkInterface anInterface : getInterfaces()) {
-            assertThat(exception.getMessage(), containsString(anInterface.getName() + ":suffix"));
+            // virtual interfaces might pop up or disappear while the test is running, so ignore them
+            if (anInterface.isVirtual() == false) {
+                assertThat(exception.getMessage(), containsString(anInterface.getName() + ":suffix"));
+            }
         }
     }
 }

